src/main/java/net/neoforged/neoforge/common/CommonHooks.java

```diff
@@ -1360,4 +1362,30 @@ public static void onChunkUnload(PoiManager poiManager, ChunkAccess chunkAccess)
             poiManager.remove(sectionPosKey);
         }
     }
+
+    /**
+     * Registers a before and after blocks that can oxidize and de-oxidize
+     *
+     * @param before block with less oxidization
+     * @param after block with more oxidization
+     */
+    public static void registerOxidizableBlock(Block before, Block after) {
+        Objects.requireNonNull(before, "Oxidizable Block before must not be null");
+        Objects.requireNonNull(after, "Oxidizable Block after must not be null");
+
+        WeatheringCopper.NEXT_BY_BLOCK.get().put(before, after);
+    }
+
+    /**
+     * Registers a before and after blocks that can be waxed and unwaxed
+     *
+     * @param before the unwaxed block
+     * @param after the waxed block
+     */
+    public static void registerWaxableBlock(Block before, Block after) {
```

Comment:
Likewise, I feel this would fit better in its own type. Maybe we should create a `WaxableBlockRegistry` which contains this logic, and then change the patch to:
```diff
diff --git a/patches/net/minecraft/world/item/HoneycombItem.java.patch b/patches/net/minecraft/world/item/HoneycombItem.java.patch
new file mode 100644
index 0000000000..430a321653
--- /dev/null
+++ b/patches/net/minecraft/world/item/HoneycombItem.java.patch
@@ -0,0 +1,28 @@
+--- a/net/minecraft/world/item/HoneycombItem.java
++++ b/net/minecraft/world/item/HoneycombItem.java
+@@ -21,7 +_,7 @@
+ 
+ public class HoneycombItem extends Item implements SignApplicator {
+-    public static final Supplier<BiMap<Block, Block>> WAXABLES = Suppliers.memoize(
++    public static final Supplier<BiMap<Block, Block>> WAXABLES = net.neoforged.something.or.other.WaxableBlockRegistry::getWaxables;
++    public static final Supplier<BiMap<Block, Block>> VANILLA_WAXABLES = Suppliers.memoize(
+         () -> ImmutableBiMap.<Block, Block>builder()
+                 .put(Blocks.COPPER_BLOCK, Blocks.WAXED_COPPER_BLOCK)
+                 .put(Blocks.EXPOSED_COPPER, Blocks.WAXED_EXPOSED_COPPER)
+                 .put(Blocks.WEATHERED_COPPER, Blocks.WAXED_WEATHERED_COPPER)
+@@ -58,7 +_,7 @@
+                 .put(Blocks.EXPOSED_COPPER_BULB, Blocks.WAXED_EXPOSED_COPPER_BULB)
+                 .put(Blocks.WEATHERED_COPPER_BULB, Blocks.WAXED_WEATHERED_COPPER_BULB)
+                 .put(Blocks.OXIDIZED_COPPER_BULB, Blocks.WAXED_OXIDIZED_COPPER_BULB)
+                 .build()
+     );
+-    public static final Supplier<BiMap<Block, Block>> WAX_OFF_BY_BLOCK = Suppliers.memoize(() -> WAXABLES.get().inverse());
++    public static final Supplier<BiMap<Block, Block>> WAX_OFF_BY_BLOCK = net.neoforged.something.or.other.WaxableBlockRegistry::getReverseWaxables;
++    public static final Supplier<BiMap<Block, Block>> VANILLA_WAX_OFF_BY_BLOCK = Suppliers.memoize(() -> WAXABLES.get().inverse());
+ 
```

Reply:
Alternatively, if we're moving to using the data map approach we could keep this class the same and simply deprecate the fields, pointing users towards the data map.

Reply:
yeah, i'll probably create datamaps for both these and deprecate the fields

Reply:
Those fields are all just `Supplier`s -- could you not just direct the supplier towards a map derived from the datamap?